### PR TITLE
Ditch UUID for standard lib

### PR DESCRIPTION
**Pull request by scrogson**
Copy UUID generation from Ecto and remove `elixir_uuid` dependency.

### CODE REVIEW

**Comment by hassox** [comment]
I want to click on hell yes, do you think there might be any issue with folks using uuid as the data type in postgres?

**Comment by scrogson** [comment]
> do you think there might be any issue with folks using uuid as the data type in postgres?

@hassox that's a great question. I haven't given that any thought...let's get some input first and make sure it's not going to cause any trouble.

**Comment by hassox** [comment]
Looks like in GuardianDb we're using a string rather than a UUID. I'd say that's probably enough. We _might_ want to wrap it in a backwards incompatible version?

File: mix.exs [hunk 1]
```diff
@@ -44,7 +44,7 @@ defmodule Guardian.Mixfile do
   defp elixirc_paths(_), do: ["lib"]
 
   def application do
-    [applications: [:logger, :poison, :jose, :elixir_uuid]]
+    [applications: [:crypto, :logger, :poison, :jose]]
```
Comment (yordis) [inline mix.exs]: We dont need this anymore.

Use `extra_applications: [:logger]`

**Comment by yordis** [comment]
Worst case, let a config that allow you to put the `id` generator which could be UUID.

In my particular case, I am moving all the infrastructure to use UUIDs everywhere.

File: mix.exs (hunk 1 above)
Comment (scrogson) [reply mix.exs]: Yep, good point.

File: lib/guardian/uuid.ex [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Copied from https://github.com/elixir-ecto/ecto/blob/v2.2/lib/ecto/uuid.ex
```
Comment (scrogson) [inline lib/guardian/uuid.ex]: @ericmj is this sufficient?

Comment (yordis) [reply lib/guardian/uuid.ex]: Seems good to me, but why we dont use UUID package again?

Comment (ericmj) [reply lib/guardian/uuid.ex]: 👍

Comment (scrogson) [reply lib/guardian/uuid.ex]: For such little and simple code...I definitely prefer copying from Ecto than pulling in a package.

**Pull request merged**